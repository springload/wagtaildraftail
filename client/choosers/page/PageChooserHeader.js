import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  onSearch: PropTypes.func.isRequired,
  searchEnabled: PropTypes.bool.isRequired,
};

const PageChooserHeader = ({ onSearch, searchEnabled }) => (
  <header className="nice-padding hasform">
    <div className="row">
      <div className="left">
        <div className="col">
          <h1 className="icon icon-doc-empty-inverse">
            Choose a page
          </h1>
        </div>
        <form className="col search-form" noValidate={true}>
          <ul className="fields">
            <li className="required">
              <div className="field char_field text_input field-small iconfield">
                <label htmlFor="id_q">
                  Search term:
                </label>
                <div className="field-content">
                  <div className="input icon-search ">
                    <input
                      onChange={e => onSearch(e.target.value)}
                      placeholder="Search"
                      type="text"
                      disabled={!searchEnabled}
                    />
                    <span />
                  </div>
                </div>
              </div>
            </li>
            <li className="submit visuallyhidden">
              <input value="Search" className="button" type="submit" />
            </li>
          </ul>
        </form>
      </div>
      <div className="right" />
    </div>
  </header>
);

PageChooserHeader.propTypes = propTypes;

export default PageChooserHeader;
