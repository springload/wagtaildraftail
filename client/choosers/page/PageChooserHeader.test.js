import React from 'react';
import { shallow } from 'enzyme';
import PageChooserHeader from './PageChooserHeader';

describe('PageChooserHeader', () => {
  it('renders', () => {
    expect(shallow((
      <PageChooserHeader
        onSearch={() => {}}
        searchEnabled={false}
      />
    ))).toMatchSnapshot();
  });
});
