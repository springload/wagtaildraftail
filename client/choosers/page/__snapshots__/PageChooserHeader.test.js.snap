// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageChooserHeader renders 1`] = `
<header
  className="nice-padding hasform"
>
  <div
    className="row"
  >
    <div
      className="left"
    >
      <div
        className="col"
      >
        <h1
          className="icon icon-doc-empty-inverse"
        >
          Choose a page
        </h1>
      </div>
      <form
        className="col search-form"
        noValidate={true}
      >
        <ul
          className="fields"
        >
          <li
            className="required"
          >
            <div
              className="field char_field text_input field-small iconfield"
            >
              <label
                htmlFor="id_q"
              >
                Search term:
              </label>
              <div
                className="field-content"
              >
                <div
                  className="input icon-search "
                >
                  <input
                    disabled={true}
                    onChange={[Function]}
                    placeholder="Search"
                    type="text"
                  />
                  <span />
                </div>
              </div>
            </div>
          </li>
          <li
            className="submit visuallyhidden"
          >
            <input
              className="button"
              type="submit"
              value="Search"
            />
          </li>
        </ul>
      </form>
    </div>
    <div
      className="right"
    />
  </div>
</header>
`;
